src/lxc/cmd/lxc_usernsexec.c

```diff
@@ -228,8 +229,20 @@ static int read_default_map(char *fnam, int which, char *user)
 			return -1;
 		}
 
-		newmap->hostid = atol(p1 + 1);
-		newmap->range = atol(p2 + 1);
+		ret = lxc_safe_ulong(p1 + 1, &newmap->hostid);
+		if (ret < 0) {
+			fclose(fin);
+			free(line);
```

Comment:
Like the `!tmp` case below this should also `free(newmap)` - perhaps we should have an `int result` on top and just `break` here (and set it to `0` before the break at the loop's bottom) - or the usual goto...